tests/core/dom/range/getclientrects.js

```diff
@@ -0,0 +1,236 @@
+					0: {
+						right: 426,
+						bottom: 222,
+						left: 263,
+						top: 142,
+						width: 163,
+						height: 80
+					}
+				}
+			} );
+		},
+
+		'test three line selection': function() {
+			this._assertRectList( 'three-line-selection', {
+				defaultExpected: {
+					0: {
+						x: 426,
+						y: 100,
+						width: 163,
+						height: 61,
+						top: 100,
+						right: 589,
+						bottom: 161,
+						left: 426
+					},
+					1: {
+						x: 100,
+						y: 161,
+						width: 163,
+						height: 61,
+						top: 161,
+						right: 263,
+						bottom: 222,
+						left: 100
+					},
+					2: {
+						x: 263,
+						y: 161,
+						width: 163,
+						height: 61,
+						top: 161,
+						right: 426,
+						bottom: 222,
+						left: 263
+					},
+					3: {
+						x: 426,
+						y: 161,
+						width: 163,
+						height: 61,
+						top: 161,
+						right: 589,
+						bottom: 222,
+						left: 426
+					},
+					4: {
+						x: 100,
+						y: 222,
+						width: 163,
+						height: 61,
+						top: 222,
+						right: 263,
+						bottom: 283,
+						left: 100
+					}
+				},
+				ie: {
+					0: {
+						width: 163,
+						height: 61,
+						top: 100,
+						right: 589,
+						bottom: 161,
+						left: 426
+					},
+					1: {
+						width: 489,
+						height: 61,
+						top: 161,
+						right: 589,
+						bottom: 222,
+						left: 100
+					},
+					2: {
+						width: 163,
+						height: 61,
+						top: 222,
+						right: 263,
+						bottom: 283,
+						left: 100
+					}
+				},
+				polyfill: {
+					0: {
+						right: 426,
+						bottom: 283,
+						left: 263,
+						top: 142,
+						width: 163,
+						height: 141
+					}
+				}
+			} );
+		},
+
+
+		_selectFixture: function( fixtureId ) {
+			var range = bender.tools.range.setWithHtml( this.playground, doc.getById( fixtureId ).getHtml().replace( /[\t\n]/g, '' ) );
+
+			doc.getSelection().selectRanges( [ range ] );
+
+			return range;
+		},
+
+		_assertRectList: function( fixtureId, expectedMap ) {
+			var range = this._selectFixture( fixtureId ),
+				rects = range.getClientRects(),
+				expectedKey = 'defaultExpected',
+				expectedRects,
+				curExpectedRect;
+
+			if ( typeof document.getSelection !== 'function' ) {
+				expectedKey = 'polyfill';
+			} else if ( CKEDITOR.env.ie ) {
+				expectedKey = 'ie';
+			}
+
+			expectedRects = expectedMap[ expectedKey ] || expectedMap.defaultExpected;
+
+			for ( var index in expectedRects ) {
+				if ( CKEDITOR.env.ie ) {
+					// IE and Edge doesn't return x and y properties, emulate that. (#1852)
+					if ( typeof expectedRects[ index ].x !== 'undefined' ) {
+						delete expectedRects[ index ].x;
+					}
+
+					if ( typeof expectedRects[ index ].y !== 'undefined' ) {
+						delete expectedRects[ index ].y;
+					}
+				}
+
+				for ( var key in expectedRects[ index ] ) {
+					curExpectedRect = rects[ index ][ key ];
+
+					if ( CKEDITOR.env.ie && !CKEDITOR.env.edge ) {
+						if ( rects.length === 1 && key === 'height' || key === 'top' ) {
```

Comment:
This is not equivalent of `rects.length === 1 && key === 'height' || rects.length === 1 && key === 'top'`.